filebeat/input/filestream/internal/input-logfile/harvester.go

```diff
@@ -55,6 +56,10 @@ type HarvesterGroup struct {
 
 // Run starts the Harvester for a Source.
 func (hg *HarvesterGroup) Run(ctx input.Context, s Source) error {
+	if hg.limit != 0 && uint32(len(hg.readers)) >= hg.limit {
+		return fmt.Errorf("harvester limit reached")
+	}
```

Comment:
If we return here, is there a chance to still pick up the file later? Instead of failing this looks like a typical case for a semaphore. Block until the limit gets decreased.

Reply:
That's a good point.